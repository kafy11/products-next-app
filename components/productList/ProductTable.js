import ProductRow from './ProductRow';

const ProductTable = ({ products, onDeleteProduct }) => {
    const renderTableRows = () => products.map((product, i) => (
        <ProductRow 
            key={`${i}`}
            product={product}
            onDelete={onDeleteProduct}
        />
    ));

    return (
        <table className="table">
            <thead>
                <tr>
                    <th></th>
                    <th scope="col">Product</th>
                    <th scope="col">Price</th>
                    <th scope="col">Qty.</th>
                </tr>
            </thead>
            <tbody>
                {renderTableRows()}
            </tbody>
        </table>
    );
}

export default ProductTable;