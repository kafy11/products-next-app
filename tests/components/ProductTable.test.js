import { shallow } from 'enzyme';
import ProductTable from '../../components/productList/ProductTable';
import products from '../fixtures/products';

let handleDelete, wrapper;

beforeEach(() => {
    handleDelete = jest.fn();
    wrapper = shallow(
        <ProductTable 
            products={products}
            onDeleteProduct={handleDelete} 
        />
    );
});

describe('ProductTable', () => {
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should handle onDeleteProduct', () => {
        const { product_id } = products[0];
        wrapper.find('ProductRow').first().prop('onDelete')(product_id);
        expect(handleDelete).toHaveBeenLastCalledWith(product_id);
    });
    
    it('should render a ProductRow for each product', () => {
        const rows = wrapper.find('ProductRow');
        expect(rows.length).toEqual(products.length);
    });
    
    it('should pass the product to the ProductRow', () => {
        const product = wrapper.find('ProductRow').first().prop('product');
        expect(product).toEqual(products[0]);
    });
});