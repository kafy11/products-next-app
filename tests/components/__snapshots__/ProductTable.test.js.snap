// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTable should render correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th />
      <th
        scope="col"
      >
        Product
      </th>
      <th
        scope="col"
      >
        Price
      </th>
      <th
        scope="col"
      >
        Qty.
      </th>
    </tr>
  </thead>
  <tbody>
    <ProductRow
      key="0"
      onDelete={[MockFunction]}
      product={
        Object {
          "name": "PS4",
          "price": 399,
          "product_id": 1,
          "quantity": 1,
        }
      }
    />
    <ProductRow
      key="1"
      onDelete={[MockFunction]}
      product={
        Object {
          "name": "Xbox One",
          "price": 499,
          "product_id": 2,
          "quantity": 2,
        }
      }
    />
    <ProductRow
      key="2"
      onDelete={[MockFunction]}
      product={
        Object {
          "name": "Nintendo Switch",
          "price": 299,
          "product_id": 3,
          "quantity": 3,
        }
      }
    />
  </tbody>
</table>
`;
